### PR TITLE
Request user attention when bell received in macOS

### PR DESCRIPTION
**Pull request by brennie**
We now request the user's attention in macOS when a visual bell
character is received. This manifests as the window jumping up and
down in the dock.

Depends on #1403.
Depends on tomaka/winit#664 for the changes to `WindowExt` on macOS.

### CODE REVIEW

**Comment by brennie** [comment]
I apparently had a version of this code sitting around for a *year* without opening a pull request. Oops.

**Comment by chrisduerr** [comment]
I'm not sure if Alacritty is the right place for this. As you can see with the other platforms, the `self.window.set_urgent` call is used.

It would probably be good to propose this change to winit first. See [this](https://docs.rs/winit/0.17.2/winit/struct.Window.html#method.set_urgent).

If the only effect of this change is that the window is shown, maybe just making use of the [`self.window.show`](https://docs.rs/winit/0.17.2/winit/struct.Window.html#method.show) call is a better choice.

**Comment by jwilm** [comment]
I'm open to landing this here before winit supports it. It would be good to land this upstream, though.

**Comment by brennie** [comment]
I'll open a PR with winit and update this one once that lands and is released.

@chrisduerr The window is not shown, its icon just "jumps" in the docks.

**Comment by chrisduerr** [comment]
Thanks for the clarification @brennie. Then I'd definitely push for getting this into upstream and just making use of that functionality once it's ready.

**Comment by brennie** [comment]
The only issue is `glutin` as at 0.18 which depends on `winit` 0.17 (where this would land I assume) and y'all are one `glutin` 0.16 / `winit` 0.15, so it may be a bit before this can land in that form. Happy to continue working on it though :)

**Comment by chrisduerr** [comment]
We're working on updating to glutin 0.18 here: #1403. This will be merged as soon as the remaining issues are resolved upstream, so if this is merged upstream before the glutin 0.18 PR is merged, chances are it will be pulled in with that soon.

**Comment by brennie** [comment]
I based this off the #1403 so I could use an updated winit for testing and GitHub apparently won't let you stack one PR on top of another :/ So there are going to be a bunch of duplicate commits in this PR until #1403 lands.

**Comment by chrisduerr** [comment]
#1403 has been merged. This can now be rebased on top of the latest master.

**Comment by brennie** [comment]
This has been rebased off master.

File: src/window.rs [hunk 1]
```diff
@@ -358,13 +358,20 @@ impl Window {
...
+
     #[cfg(
         not(
             any(
                 target_os = "linux",
                 target_os = "freebsd",
                 target_os = "dragonfly",
-                target_os = "openbsd"
+                target_os = "openbsd",
+                target_os = "macos"
```
Comment (chrisduerr) [inline src/window.rs]: This should be changed to `#[cfg(windows)]`, from the platforms we support, that's the only one this applies to.

**Comment by brennie** [comment]
I think I've successfully addressed all issues.

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - New configuration option `persistent_logging` and CLI flag `--persistent-logging`,
     for keeping the log file after closing Alacritty
 - `ClearLogNotice` action for removing the warning and error message
+- Terminal bells on macOS will now request the user's attention in the window.
```
Comment (chrisduerr) [inline CHANGELOG.md]: No punctuation mark should be at the end of the change log entry.

**Pull request merged**